### PR TITLE
Deb/RPM packaging for Calico 2.3.0

### PR DESCRIPTION
**Pull request by neiljerram**

### CODE REVIEW

**Comment by neiljerram** [comment]
I've created a 2.3.x-series branch here, similar to those that we have for other recent release series.

@lwr20 I think it makes sense for you to review as soon as you have time - basically just a sanity check, as the contents of the changelogs here are generated from tags that we've already generated and released.

Before merging:
- we should get greens for the semaphore and calico-test CIs above, as usual
- I also propose to test the corresponding Deb and RPM packages; so far I've just built those locally, but will push to test repos and then run through calico-test.

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,56 @@
...
+  * Felix 2.3.0 (from Git commit 85f9fff).
+    [Changes recorded in 2.3.0 tag]
+    This is a feature release of Felix, Calico's per-host agent.
+    
+    The headline feature in this release is a significant increase in scale when using the
+    Kubernetes datastore driver by introducing support for a new daemon, Typha. Typha
+    connects to the Kubernetes API server and fans out updates to a number of Felixes.
+    By having only a handful of Typha instances connected to the API server instead of
+    many Felixes, we place a lot less load on the API server.  In addition, Typha is able
+    to squash unimportant updates form the API server, significantly reducing the
```
Comment (lwr20) [inline debian/changelog]: typo: form/from

File: debian/changelog [hunk 2]
```diff
@@ -1,3 +1,56 @@
...
+    number of mesages each Felix has to handle.
```
Comment (lwr20) [inline debian/changelog]: typo: mesages/messages

File: debian/changelog [hunk 3]
```diff
@@ -1,3 +1,56 @@
...
+    
+    - Performance: Scan iptables-save output incrementally when calculating hashes.
+    - Performance: Disable WithFields logging in the ipsets resync parse loop.
+    - Performance: Disable some WithField calls in the inner iptables resync loop.
+    - Allow the iptables post-write check interval to be set.
+    - Improve diagnostics around ipset restore failures
+    - Log ip6tables-save stderr if it fails
+    - Convert interval config parameters to time.Duration for increased precision.
+    - Rev libcalico-go to v1.4.4 and Typha to v0.2.2.
+    - Make JUST_A_MINUTE the default, as it's more useful for devs
```
Comment (lwr20) [inline debian/changelog]: Isn't this part of the k8sfv functional tests?

File: debian/changelog [hunk 4]
```diff
@@ -1,3 +1,56 @@
...
+    The k8sfv functional tests also got a number of enhancements:
+    
+    - Update imports for recent client-go/apimachinery moves
+    - Check Felix does not die before end of test
+    - mechanism for only running quick tests
+    - fix client authorization to API server 1.6
+    - Ensure that ip6_tables module is loaded
+    - put Typha in the loop.
+    
+    The DockerHub and Quay.io `calico/felix` images have been updated.  The `calico/node` image
```
Comment (lwr20) [inline debian/changelog]: Do we need this bit?

File: debian/changelog [hunk 5]
```diff
@@ -1,3 +1,56 @@
...
+    based on this release will follow shortly.
+    
+    Since we're targetting Kubernetes for this release, we haven't yet updated the OpenStack
```
Comment (lwr20) [inline debian/changelog]: Needed?

File: rpm/felix.spec [hunk 6]
```diff
@@ -152,6 +152,56 @@ rm -rf $RPM_BUILD_ROOT
...
+    
+    - Performance: Scan iptables-save output incrementally when calculating hashes.
+    - Performance: Disable WithFields logging in the ipsets resync parse loop.
+    - Performance: Disable some WithField calls in the inner iptables resync loop.
+    - Allow the iptables post-write check interval to be set.
+    - Improve diagnostics around ipset restore failures
+    - Log ip6tables-save stderr if it fails
+    - Convert interval config parameters to time.Duration for increased precision.
+    - Rev libcalico-go to v1.4.4 and Typha to v0.2.2.
+    - Make JUST_A_MINUTE the default, as it's more useful for devs
```
Comment (lwr20) [inline rpm/felix.spec]: Isn't this a k8sfv feature?

File: rpm/felix.spec [hunk 7]
```diff
@@ -152,6 +152,56 @@ rm -rf $RPM_BUILD_ROOT
...
+    - Check Felix does not die before end of test
+    - mechanism for only running quick tests
+    - fix client authorization to API server 1.6
+    - Ensure that ip6_tables module is loaded
+    - put Typha in the loop.
+    
+    The DockerHub and Quay.io `calico/felix` images have been updated.  The `calico/node` image
+    based on this release will follow shortly.
+    
+    Since we're targetting Kubernetes for this release, we haven't yet updated the OpenStack
```
Comment (lwr20) [inline rpm/felix.spec]: Needed?

File: rpm/felix.spec [hunk 8]
```diff
@@ -152,6 +152,56 @@ rm -rf $RPM_BUILD_ROOT
...
+    The k8sfv functional tests also got a number of enhancements:
+    
+    - Update imports for recent client-go/apimachinery moves
+    - Check Felix does not die before end of test
+    - mechanism for only running quick tests
+    - fix client authorization to API server 1.6
+    - Ensure that ip6_tables module is loaded
+    - put Typha in the loop.
+    
+    The DockerHub and Quay.io `calico/felix` images have been updated.  The `calico/node` image
```
Comment (lwr20) [inline rpm/felix.spec]: Needed?

**Comment by neiljerram** [comment]
@lwr20 Thanks for the careful review.  As we've since discussed, all of the issues above exist because they are already present in the content of the Git tag 2.3.0 that we've already released (and which you can see with `git show 2.3.0`).  Some of them have since been corrected in the edited view of that content at https://github.com/projectcalico/felix/releases/tag/2.3.0, but it appears that that does not change the original Git tag content.

Given that this is for a release, 2.3.0, that is already old now, I don't think it's worth correcting these issues in the deb/rpm changelogs.  For 2.4.0, however, I _will_ review the process so that it is possible to review, catch and fix issues like these before publishing new packages for testing.

**Pull request merged**